android/src/test/java/com/segment/analytics/kotlin/android/AndroidLifecyclePluginTests.kt

```diff
@@ -54,13 +62,19 @@ class AndroidLifecyclePluginTests {
 
     @Before
     fun setup() {
-        analytics = Analytics(
+        analytics = spyk(Analytics(
             Configuration(
                 writeKey = "123",
                 application = mockContext,
                 storageProvider = AndroidStorageProvider
             )
-        )
+        ))
+        every { analytics.analyticsScope } returns testScope
+        every { analytics.analyticsDispatcher } returns testDispatcher
+        every { analytics.networkIODispatcher } returns testDispatcher
+        every { analytics.fileIODispatcher } returns testDispatcher
+        every { analytics.store } returns spyStore(testScope, testDispatcher)
+        lifecyclePlugin = AndroidLifecyclePlugin()
```

Comment:
any reason why this needs to be initialized here vs how it was before

Reply:
yeah. this does not have to be here. I will revert this change. I was blindly trying to fix the flaky test. this eliminated a possible cause, since `AndroidLifecyclePlugin` manages some states internally.